libs/base/Decidable/Equality.idr

```diff
@@ -123,16 +107,16 @@ implementation (DecEq a, DecEq b) => DecEq (a, b) where
 lemma_val_not_nil : (the (List _) (x :: xs) = Prelude.Nil {a = t} -> Void)
```

Comment:
All of these names are pretty terrible and the lemmas not that
useful given they're direct corollaries of the fact that `(::)`
is injective.

They're not exported so we can maybe get rid of all of this
boilerplate & replace them with `p . (fst / snd) . consInjective`
in the `decEq` proof for list.

Reply:
Thanks, I couldn't figure out how to get that to work before.